.github/workflows/test.yml

```diff
@@ -94,3 +93,78 @@ jobs:
 
       - name: Perform CodeQL Analysis
         uses: github/codeql-action/analyze@v2
+
+  e2e_tests:
+    needs: unit_tests
```

Comment:
Add a condition `(github.event.pull_request.head.repo.full_name || github.push.repository.full_name) == github.repository` so it's not run against forks and maybe draft PR. 

I would also consider running it after quality gates passed, so that it's the final test of the workflow.

Reply:
Done